### PR TITLE
Source Apify Dataset: fix broken stream, manifest refactor

### PR DESCRIPTION
**Pull request by vdusek**
## What

- Remove the old broken Item Collection stream.
    - The schema of Apify Dataset is at least Actor-specific, so we cannot have a general Stream for getting data from Dataset.
- Add a new Item Collection (WCC) stream - specific for the Website Content Crawler Actor.
- Basically, complete manifest refactor, simplification, merge `spec.yaml` and `manifest.yaml`, ...

## How

- Updated manifests, executed checks & tests locally, imported the manifest into the local Airbyte instance a tried to execute the Streams. Everything works fine.

## Recommended reading order

1. manifest
2. schemas
3. others

## 🚨 User Impact 🚨

- Streams were renamed to match Apify API endpoints. 
- Item Collection stream was removed since it does not work (the Dataset endpoint does not have static schema)
- Add a new Item Collection WCC stream for getting data produced by Website Content Crawler (WCC) Actor.

## Pre-merge Actions

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-apify-dataset/metadata.yaml [hunk 1]
```diff
@@ -23,6 +23,9 @@ data:
       1.0.0:
         upgradeDeadline: 2023-08-30
         message: "Update spec to use token and ingest all 3 streams correctly"
+    1.1.0:
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-apify-dataset/metadata.yaml:26]: 2.0.0

File: airbyte-integrations/connectors/source-apify-dataset/source_apify_dataset/schemas/item_collection_wcc.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+          "type": "number"
+        },
+        "loadedTime": {
+          "type": "string"
+        },
+        "loadedUrl": {
+          "type": "string"
+        },
+        "referrerUrl": {
+          "type": "string"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-apify-dataset/source_apify_dataset/schemas/item_collection_wcc.json:22]: Please use the pattern ["null", "string"]

File: airbyte-integrations/connectors/source-apify-dataset/source_apify_dataset/schemas/item_collection_wcc.json [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        }
+      },
+      "type": "object"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-apify-dataset/source_apify_dataset/schemas/item_collection_wcc.json:25]: Add `additionalProperties: true` check https://docs.airbyte.com/connector-development/schema-reference

File: airbyte-integrations/connectors/source-apify-dataset/source_apify_dataset/manifest.yaml [hunk 4]
```diff
@@ -1,109 +1,122 @@
...
+        type: string
+        title: Dataset ID
+        description: >-
+          ID of the dataset you would like to load to Airbyte. In Apify Console, you can view your datasets in the
+          <a href="https://console.apify.com/storage/datasets">Storage section under the Datasets tab</a>
+          after you login. See the <a href="https://docs.apify.com/platform/storage/dataset">Apify Docs</a>
+          for more information.
+        examples:
+          - rHuMdwm6xCFt6WiGU
+      clean:
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-apify-dataset/source_apify_dataset/manifest.yaml:36]: This is not used anywhere.

**Review by marcosmarxm — DISMISSED** [review]
Left some comments.

**Comment by vdusek** [comment]
Thanks for the review @marcosmarxm , the code is updated.

File: airbyte-integrations/connectors/source-apify-dataset/integration_tests/configured_catalog.json [hunk 5]
```diff
@@ -2,7 +2,7 @@
   "streams": [
     {
       "stream": {
-        "name": "datasets",
+        "name": "Dataset Collection",
```
Comment (flash1293) [inline airbyte-integrations/connectors/source-apify-dataset/integration_tests/configured_catalog.json:5]: @vdusek Streams gets mapped to tables in most destinations - spaces etc. get normalized but it keeps things reliable if they have a "table"-y name already.

What do you think about naming them `dataset_collection`, `dataset` and `item_collection` and explaining what they mean in the documentation

**Comment by flash1293** [comment]
Besides what's mentioned above, the acceptance tests need to be adjusted: https://github.com/airbytehq/airbyte/blob/c0e2f33e08e6b0ee0e4aa395e100b06e5f80a2a9/airbyte-integrations/connectors/source-apify-dataset/acceptance-test-config.yml#L35

* Add an exception for JSON schema backwards compatible check
* Adjust stream names for the full refresh checks

File: airbyte-integrations/connectors/source-apify-dataset/integration_tests/configured_catalog.json (hunk 5 above)
Comment (vdusek) [reply airbyte-integrations/connectors/source-apify-dataset/integration_tests/configured_catalog.json:5]: Makes sense, updated, thanks.

**Comment by vdusek** [comment]
@flash1293 Thanks for the review, everything should be resolved except for the following...

> Add an exception for JSON schema backwards compatible check

I don't know what you mean by that, could you please explain that in more detail? Thanks.

**Comment by flash1293** [comment]
Thanks for the changes @vdusek 

> Add an exception for JSON schema backwards compatible check
I don't know what you mean by that, could you please explain that in more detail? Thanks.

I was referring to the acceptance tests but no worry, I will take care of that

**Comment by flash1293** [comment]
Thanks @vdusek 

I fixed a few problems, there is one thing remaining:
> The item_collection_website_content_crawler stream has the following schema errors:
> Additional properties are not allowed ('#debug', '#error', 'pageTitle' were unexpected)

Do you think we should sync these properties (if yes we should add them to the schema)? If not we can also remove them with a remove transformation in the manifest (like here: https://github.com/airbytehq/airbyte/blob/f5c7c1c0b8dc461f64e8af3f8584475d8a887cfe/airbyte-integrations/connectors/source-plaid/source_plaid/manifest.yaml#L93 )

**Comment by vdusek** [comment]
> Do you think we should sync these properties (if yes we should add them to the schema)? If not we can also remove them with a remove transformation in the manifest (like here:

Hi, in the WCC datasets, there should be no properties such as `"#debug"`, `"#error"` or `"pageTitle"`.

Here is an example of the WCC dataset:
```json
[
  {
    "url": "https://docs.apify.com/academy/web-scraping-for-beginners",
    "crawl": {
      "loadedUrl": "https://docs.apify.com/academy/web-scraping-for-beginners",
      "loadedTime": "2023-09-08T15:11:20.522Z",
      "referrerUrl": "https://docs.apify.com/academy/web-scraping-for-beginners",
      "depth": 0,
      "httpStatusCode": 200
    },
    "metadata": {
      "canonicalUrl": "https://docs.apify.com/academy/web-scraping-for-beginners",
      "title": "Web scraping for beginners | Academy | Apify Documentation",
      "description": "Learn how to develop web scrapers with this comprehensive and practical course. Go from beginner to expert, all in one place.",
      "author": null,
      "keywords": null,
      "languageCode": "en"
    },
    "screenshotUrl": null,
    "text": "Web scraping for beginners...",
    "markdown": "## Web scraping for beginners..."
  },
  {
    "url": "https://docs.apify.com/academy/web-scraping-for-beginners/introduction",
    "crawl": {
      "loadedUrl": "https://docs.apify.com/academy/web-scraping-for-beginners/introduction",
      "loadedTime": "2023-09-08T15:11:32.622Z",
      "referrerUrl": "https://docs.apify.com/academy/web-scraping-for-beginners",
      "depth": 1,
      "httpStatusCode": 200
    },
    "metadata": {
      "canonicalUrl": "https://docs.apify.com/academy/web-scraping-for-beginners/introduction",
      "title": "Introduction | Academy | Apify Documentation",
      "description": "Start learning about web scraping, web crawling, data extraction, and popular tools to start developing your own scraper.",
      "author": null,
      "keywords": null,
      "languageCode": "en"
    },
    "screenshotUrl": null,
    "text": "Introduction\nStart learning about web scraping, ..."
  },
  "...": "..."
]
```

However, these fields are in the old schema ([source_apify_dataset/schemas/item_collection.json](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-apify-dataset/source_apify_dataset/schemas/item_collection.json)). So it might have something to do with it.

**Review by flash1293 — APPROVED** [review]
@vdusek I changed the acceptance tests to actually run on a WCC dataset. I had to adjust the schemas a bit to make it work (some fields were missing in the provided schemas), but it's passing now

**Pull request merged**